libweb3jsonrpc/Test.cpp

```diff
@@ -23,16 +23,39 @@
 #include "Test.h"
 #include <jsonrpccpp/common/errors.h>
 #include <jsonrpccpp/common/exception.h>
-#include <libethereum/ClientTest.h>
+#include <libdevcore/CommonJS.h>
 #include <libethereum/ChainParams.h>
+#include <libethereum/ClientTest.h>
 
 using namespace std;
 using namespace dev;
+using namespace dev::eth;
 using namespace dev::rpc;
 using namespace jsonrpc;
 
 Test::Test(eth::Client& _eth): m_eth(_eth) {}
 
+string exportLog(eth::LogEntries const& _logs)
+{
+    RLPStream s;
+    s.appendList(_logs.size());
+    for (eth::LogEntry const& l : _logs)
+        l.streamRLP(s);
+    return toHexPrefixed(sha3(s.out()));
+}
+
+string Test::test_getLogHash(string const& _txHash)
+{
+    if (m_eth.blockChain().isKnownTransaction(h256(_txHash)))
+    {
+        LocalisedTransaction t = m_eth.localisedTransaction(h256(_txHash));
+        BlockReceipts const& blockReceipts = m_eth.blockChain().receipts(t.blockHash());
+        if (blockReceipts.receipts.size() != 0)
+            return exportLog(blockReceipts.receipts[t.transactionIndex()].log());
+    }
+    return toJS(dev::EmptyListSHA3);
```

Comment:
Why should it be empty list in case transaction not found? This way the user of the method can't understand whether the transaction produced no log or it doesn't exist at all.

Maybe we should return error in case transaction not found?

Reply:
I think returning error is cleaner than magic value of hash.
You should also be able to make the error more specific by adding error message like `throw JsonRpcException(Errors::ERROR_RPC_INVALID_PARAMS, "Transaction not found")`

Reply:
Or you could introduce new error code and return it like `throw JsonRpcException(TxNotFoundErrorCode)`

Maybe check what geth returns from `eth_getTransactionByHash` in this case

Reply:
ideally all json requests response should have a common scheme that all clients follow.